### ISSUE TITLE
Show reports from header files

### ISSUE THREAD

**Issue by csordasmarton** [Ericsson/CodecheckerVSCodePlugin#10]
The plugin shows reports only from source files which have a plist file and not from header files.

Example:
`main.cpp`:
```c++
#include "main.h"

int main() {
  sizeof(42);
  sizeof(43);
  return foo(0);
}
```
`main.h`:
```h
double foo(int param)
{
  return 1 / param;
}
```
I analyzed it to a .codechecker directory to my workspace, build the plugin and opened the `main.cpp` file and opened the `CodeChecker view`.
- `In related files` section we can see that there is a bug in the `main.h` header file:
![image](https://user-images.githubusercontent.com/6695818/127624372-e88bdcf9-2e09-4fca-b393-4d70beee4cb0.png)
- If I open the `main.h` header file the CodeChecker view will not show any reports but on the `main.cpp` file but we saw previously that we have a division by zero error in this file:
![image](https://user-images.githubusercontent.com/6695818/127624648-ca1088d1-953a-4a55-a9b2-fa369a1d9f55.png)

This way we are hiding bugs from the user which is not good.

**Comment by Discookie** [comment]
Quote from https://github.com/Ericsson/CodecheckerVSCodePlugin/pull/9#discussion_r679884900:

The 'Related files' section displays errors in headers and such, but they are in the currently opened file's .plist file.

To keep the errors when changing files, either:

* the source-file needs to be tracked (via eg. DiagnosticsAPI.activeReport), which only provides one previous files' error,
* all the reports need to be pre-processed, to see which report contains errors for the file,
* or the found diagnostics need to be sorted and kept loaded at runtime.

Originally I implemented the first option, and it resulted in very inconsistent behavior.
The third option shouldn't be too difficult to implement, but it isn't straightforward either, and the results either need to be cached, or the displayed errors will be inconsistent when the user restarts their editor.

**Comment by csordasmarton** [comment]
This was fixed in #37.
